File: SPOJ/Random/GSS6/brute.py
n = input()
a = [int(x) for x in raw_input().split()]

q = input()
for _ in xrange(q):
	qr = raw_input().split()
	for i in xrange(1, len(qr)):
		qr[i] = int(qr[i])

	x = qr[1]	
	if qr[0] == 'I':
		x -= 1
		a = a[:x] + [qr[2]] + a[x:]
	elif qr[0] == 'D':
		x -= 1
		a = a[:x] + a[x+1:]
	elif qr[0] == 'R':
		x -= 1
		a[x] = qr[2]
	else:
		x = qr[1] - 1
		y = qr[2] - 1

		mx = max(a[x:y+1])
		sm = 0
		
		for i in xrange(x,y+1):
			sm += a[i]
			if sm < 0:
				sm = 0
			else:
				mx = max(mx, sm)
		
		print mx
		

File: CodeChef/SNACKDOWN/16/Qualifier/Problem B/brute.cc
#include <bits/stdc++.h>

#define sd(x) scanf("%d",&x)
#define sd2(x,y) scanf("%d%d",&x,&y)
#define sd3(x,y,z) scanf("%d%d%d",&x,&y,&z)

#define fi first
#define se second
#define pb push_back
#define mp make_pair
#define foreach(it, v) for(__typeof((v).begin()) it=(v).begin(); it != (v).end(); ++it)
#define meta __FUNCTION__,__LINE__

#define _ ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
#define __ freopen("input.txt","r",stdin);freopen("output.txt","w",stdout);

using namespace std;

template<typename S, typename T> 
ostream& operator<<(ostream& out,pair<S,T> const& p){out<<'('<<p.fi<<", "<<p.se<<')';return out;}

template<typename T>
ostream& operator<<(ostream& out,vector<T> const& v){
int l=v.size();for(int i=0;i<l-1;i++)out<<v[i]<<' ';if(l>0)out<<v[l-1];return out;}

void tr(){cout << endl;}
template<typename S, typename ... Strings>
void tr(S x, const Strings&... rest){cout<<x<<' ';tr(rest...);}

typedef long long ll;
typedef pair<int,int> pii;

const long double PI = 3.1415926535897932384626433832795;

const int N = 100100;
const ll INF = (1ll << 60);

int n;
int a[N];
ll dp[N];

void solve(){
	sd(n);
	
	bool all_neg = true;
	
	for(int i = 1; i <= n; i++){
		sd(a[i]);
		if(a[i] > 0) all_neg = false;
	}
	
	ll ans = -INF;
	
	if(all_neg){
		for(int i = 1; i <= n; i++){
			ans = max(ans, 1ll*a[i]);
		}
		printf("%lld\n", ans);
		return;
	}
	
	for(int i = 1; i <= n; i++){
		ll sum = 0, mn = 0;
		for(int j = i; j <= n; j++){
			sum += a[j];
			mn = min(mn, 1ll*a[j]);
			ans = max(ans, sum - mn);
		}
	}
	
	printf("%lld\n", ans);
	
}

int main(){
	int t; sd(t);
	while(t--) solve();
	return 0;
}
